Tests/TestingApps/PaywallsTester/PaywallsTester/Data/ApplicationData.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  ApplicationData.swift
+//
+//
+//  Created by Nacho Soto on 12/11/23.
+//
+
+import Foundation
+
+import OSLog
+
+@Observable
+final class ApplicationData {
+
+    private(set) var authentication: Authentication = .unknown {
+        didSet {
+            Self.logger.info("Changed authentication: \(String(describing: self.authentication))")
+        }
+    }
+
+    @MainActor
+    func loadApplicationData() async throws {
+        self.authentication = .unknown
+
+        do {
+            self.authentication = .signedIn(try await self.manager.loadApplicationData())
```

Comment:
does this make the network call run on the main thread? I know the actor is only here, and then we do another try await inside and yet another one inside of that, I'm not entirely sure how that'll translate to swift concurrency model

Reply:
`ApplicationManager.loadApplicationData()` will be run on a background thread, because that method is not isolated to a particular actor and so it runs on a background thread chosen by the os. I did clarify my understanding of this, this Q/A was useful https://forums.swift.org/t/determining-whether-an-async-function-will-run-on-the-main-actor/60749/2

Reply:
cool thanks!